### PR TITLE
buildbot: 0.9.0rc2 -> 0.9.0rc3

### PR DESCRIPTION
**Pull request by nand0p**
- Upgrades buildbot to latest release candidate
- Binaries tested on linux
- fixes python path issue with wrapper

### CODE REVIEW

File: pkgs/development/interpreters/python/wrap.sh [hunk 1]
```diff
@@ -65,7 +65,8 @@ wrapPythonProgramsIn() {
                 # above. The script will set PYTHONPATH and PATH variables.!
                 # (see pkgs/build-support/setup-hooks/make-wrapper.sh)
                 local -a wrap_args=("$f"
-                                 --prefix PATH ':' "$program_PATH")
+                                    --prefix PYTHONPATH ':' "$program_PYTHONPATH"
+                                    --prefix PATH ':' "$program_PATH")
```
Comment (FRidh) [inline pkgs/development/interpreters/python/wrap.sh]: This change will have to be discussed in a separate PR.

**Review by FRidh — CHANGES_REQUESTED** [review]
The change to the wrapper will need to be discussed in a separete PR. Otherwise, this looks good.

File: pkgs/development/tools/build-managers/buildbot/plugins.nix [hunk 2]
```diff
@@ -5,21 +5,17 @@
...
     src = fetchurl {
-      url = "https://pypi.python.org/packages/95/47/1fef931d410cc24127564c2e193e7c1c184f5c5f481930f77c6d6840cfab/${name}.tar.gz";
-      sha256 = "01wc9bmqq1rfayqnjm7rkjhbcj7h6ah4vv10s6hglnq9s4axvxp6";
+      url = "mirror://pypi/b/${pname}/${name}.tar.gz";
+      sha256 = "0a05dgc5nn6sx3nicfvqwar9iy8yxhq92688dhs5p6mzjxaprlxm";
     };
 
     propagatedBuildInputs = with pythonPackages; [ setuptools ];
 
-    # doesn't seem to break without this...
```
Comment (ryansydnor) [inline pkgs/development/tools/build-managers/buildbot/plugins.nix]: exciting!

File: pkgs/development/tools/build-managers/buildbot/default.nix [hunk 3]
```diff
@@ -7,10 +7,10 @@
...
   pname = "buildbot";
-  version = "0.9.0rc2";
+  version = "0.9.0rc3";
   src = fetchurl {
     url = "mirror://pypi/b/${pname}/${name}.tar.gz";
-    sha256 = "06xzbjzji3by4hldwr850fc989ymsfl2s8nil21klv2g00dgmcmk";
+    sha256 = "18238n9l0pfivb23csy5lnkx91fcm4jgmhxn03f2nfnp00rlppzb";
   };
 
   buildInputs = with pythonPackages; [
```
Comment (ryansydnor) [inline pkgs/development/tools/build-managers/buildbot/default.nix]: i believe buildbot fixed their packaging so we can remove lines 62-69 here.

(https://github.com/buildbot/buildbot/blob/90ae62b19db764b8099d308abc1f6d313b0b01f3/master/docs/relnotes/index.rst#fixes)

File: pkgs/development/interpreters/python/wrap.sh (hunk 1 above)
Comment (nand0p) [inline pkgs/development/interpreters/python/wrap.sh]: moved this commit to separate PR: https://github.com/NixOS/nixpkgs/pull/18667

File: pkgs/development/tools/build-managers/buildbot/default.nix (hunk 3 above)
Comment (nand0p) [inline pkgs/development/tools/build-managers/buildbot/default.nix]: removed.
